akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -244,11 +244,16 @@ object ActorMaterializerSettings {
     outputBurstLimit:            Int,
     fuzzingMode:                 Boolean,
     autoFusing:                  Boolean,
-    maxFixedBufferSize:          Int) =
+    maxFixedBufferSize:          Int) = {
+    // these sins were comitted in the name of bin comp:
+    val config = ConfigFactory.load()
```

Comment:
This is wrong. This will load "application.conf", but the ActorSystem might not have been started with application.conf at all. If you don't want to hardcode the default values, which is what we have previously done when adding new settings it should at least be `ConfigFactory.defaultReference`. That is also not completely correct since the ActorSystem could have been started with another classloader.

Reply:
`load()` is used in several places in this file, please fix all

Reply:
Ok, I thought it was fishy that it was done like that (from the stream refs I think) but couldn't figure out a better way to do it.

Reply:
Yeah we screwed up a bit by exposing constructors that take params, and not the config. Such constructors should be deprecated and we should stick to passing the config as the "main" constructor.

Reply:
this must be fixed